# Kintamieji

Programavime kintamieji yra kaip konteineriai, kurie laiko duomenis ar reikšmes. Jie leidžia jums saugoti ir manipuliuoti informacija jūsų programoje. „Python“ yra dinamiškai rašoma kalba, tai reiškia, kad jums nereikia aiškiai nurodyti kintamojo tipo.

Norėdami sukurti kintamąjį „Python“, paprasčiausiai pasirenkate jam pavadinimą ir suteikiate jam reikšmę naudodami lygybės ženklą (=). Štai pavyzdys:


In [1]:
age = 25    # nustatoma kintamojo (intedžerio/sveikojo skaičiaus reikšmė)
print(age)  # išspausdinama, kintamojo "age" reikšmė

25


Šiame pavyzdyje, `age` yra kintamojo pavadinimas, ir `25` yra jam priskirta reikšmė

### Greita Užduotis 1

Priskirkite kintamajam teigiamo skaičiaus skaičiaus reikšmę, tada jį išspausdinkite


In [2]:
skaičius = 1
print(skaičius)

1


In [3]:
metai = 2023
print(metai)

2023


In [4]:
paukštis = "žąsis"
print(paukštis)

žąsis


---

## Kintamūjų Tipai


"Python" leidžia naudoti skirtingus duomenų tipus kintamiesiems, tokiems kaip "integer" (sveikieji skaičiai), "float" skaičiai (slankiojo kablelio skaičiai), "strings" (teksto eilutė), loginės reikšmės (booleans) (teisinga/True arba klaidinga/False) ir kt. Kintamojo tipas nustatomas pagal jo turimą reikšmę:

In [5]:
# Sveikasis skaičius (integer) arba sveikasis kintamasis
age = 25
print(age)

25


In [5]:
# Slankiojo kablelio skaičius (float)
height = 1.75
print(height)

1.75


In [6]:
# Teksto eilutės (string) kintamasis
first_name = "John"     # note: underscore is OK here.
print(first_name)

John


In [7]:
# Loginės reikšmės (boolean) kintamasis
is_student = False
print(is_student)

False


Kintamūjų reikšmė, gali būti bet kada pakeista jam priskiriant naują reikšmę, Pavyzdžiui:

In [8]:
age = 25        # pirminis skaičius
age = age + 1   # jis atnaujinamas pridedant +1
print(age)      # išspausdins: 26

26


Taip pat galite atlikti operacijas su kintamaisiais, sujungti juos į išraiškas ir naudoti jų reikšmes skaičiavimuose. Taip pat galite iš naujo priskirti kintamiesiems skirtingas reikšmias, pavyzdžiui:


In [9]:
variable = 5
print(variable)
variable = "not a number"
print(variable)

5
not a number


Tačiau tai nepatartina praktikuoti, net jei ji veikia, nes kodą bus sunkiau suprasti ir prižiūrėti. Tačiau kartais tai gali būti neišvengiama.

###  Greita Užduotis 2


Priskirkite kintamajam skirtingas reikšmes ir jas išspausdinkite

💡 Naudingas Patarimas: `print()` funcija priima ne viena kableliu atskirtą argumentą

In [10]:
kintamasis = 9
print(kintamasis)
kintamasis = "nera skaicius"
print(kintamasis)

9
nera skaicius


In [11]:
bananai = 3
obuoliai = "obuoliai"
yra_labai_skanūs = True
print(bananai, obuoliai, yra_labai_skanūs, sep=', ')

3, obuoliai, True


---

## Kintamųjų pakartotinis priskyrimas



Kintamieji gali būti priskiriami kitiems kintamiesiems, štai jums pavyzdys:

In [12]:
number1 = 5
number2 = number1
print(number1, number2)
# abiejų kintamųjų reikšmė 5

5 5


In [13]:
number1 = 10
print(number1, number2)
# nors number1 yra priskirta 10, number2 vistiek bus 5

10 5


In [14]:
number1 = 10
number2 = 20
print(number1, number2)

10 20


### Greita užduotis 3


Nustatykite bet kurią kintamojo reikšmę, tada priskirkite ją kitam kintamajam ir pakeiskite senojo reikšmę. Spausdinkite abu kintamuosius prieš ir po pakartotinio priskyrimo.

In [15]:
number1 = 10
number2 = 20
print(number1, number2)
number1 = number2
print(number1, number2)
number3 = 30
number4 = 40
print(number1, number2, number3, number4)
number3 = number4
number5 = 50
print(number1, number2, number3, number4, number5)

10 20
20 20
20 20 30 40
20 20 40 40 50


---

## Numeriniai (Skaitiniai) Kintamųjų Tipai

"Python" programavimo kalboje , intedžeriai (`int`) and slankiojo kablelio skaičiai (`float`) yra du pagrindiniai skaitinių kintamųjų tipai. Jie skiriasi savo skaičių vaizdavimu ir tvarkymu.

### Integer (`int`)

'Integer' yra sveikieji skaičiai be dešimtainių. Jie gali būti teigiami ir neigiami.

In [16]:
age = 25                # Teigiamas intedžeris
temperature = -10       # Neigiamas intedžiaris
students_count = 100    # Dar vienas teigiamas intedžeris
print(age, temperature, students_count)

25 -10 100



Anksčiau pateiktuose pavyzdžiuose `age` reiškia asmens amžių, `temperature` – neigiamą temperatūros reikšmę, o `students_count` – studentų skaičių.

### Slankiojo Kablelio Skaičiai (`float`)

Slankiojo kablelio skaičiai (float) arba frakcinės dalys. Jie gali būti teigiami ir neigiami.

In [17]:
weight = 64.5               # Pozityvus slankiojo kablelio skaičius
PI = 3.14159                # Pozityvus slankiojo kablelio skaičius
bank_balance = -2500.75     # Negatyvus slankiojo kablelio skaičius
print(weight, PI, bank_balance)

64.5 3.14159 -2500.75


Šiame pavyzdyje `weight` reiškia kieno nors svorį kilogramais, `PI` – matematinę konstantą π (pi), o `bank_balance` reiškia neigiamą banko likutį.

Sveikieji kintamieji gali reikšti tik sveikus skaičius, o slankiųjų (dešimtainių) kintamieji siūlo didesnį tikslumą ir gali rodyti trupmenines dalis.

Labai svarbu pasirinkti tinkamą kintamojo tipą (int arba float), atsižvelgiant į skaičiaus, su kuriuo dirbate, pobūdį, kad būtų užtikrintas tikslus skaičiavimas ir tinkamas duomenų pateikimas.

Nors slankiojo kablelio skaičiai (float) yra lankstesni, operacijos su sveikaisiais skaičiais turi daug didesnį našumą. Ypač iteracijose.

### Greita Užduotis 4

Parašykite ir išspausdinkite kelis kintamuosius su int ir float.

In [18]:
age = 69
address = 777
house = 666
print(age, address, house)
age = address
address = house
print(age, address, house)

69 777 666
777 666 666


---

## Kintamojo Tipo Patikrinimas

Patikrinti kintamojo tipą "Python", galima panaudoti vietinę funkciją `type()`. `type()` funkcija nurodo kintamojo arba objekto tipą.

Ją galima panaudoti taip:

In [19]:
age = 25
print(type(age))        # Išspausdina: <class 'int'>

<class 'int'>


In [20]:
weight = 64.5
print(type(weight))     # Išspausdina: <class 'float'>

<class 'float'>


In [21]:
first_name = "John"
print(type(first_name))       # Išspausdina: <class 'str'>

<class 'str'>


In [24]:
is_student = True
print(type(is_student)) # Išspausdina: <class 'bool'>

<class 'bool'>



Šiame pavyzdyje turime keturis skirtingų tipų kintamuosius: `age` (sveikas skaičius), `weight` (slankiojo kablelio skaičius), `name` (teksto eilutė) ir `is_student` (loginis). Kiekvienam kintamajam pritaikę funkciją `type()`ir išspausdinę rezultatą, galime nustatyti konkretų kiekvieno kintamojo tipą.

Funkcijos `type()` išvestis rodoma `<class 'typename'>` forma, kur `typename` reiškia konkretų kintamojo tipą.

Naudodami `type()` galite dinamiškai nustatyti kintamojo tipą vykdymo metu, o tai gali būti naudinga derinant arba dirbant su kodu, kuris apima skirtingus duomenų tipus.

Jei norite, kad tipo objektas būtų skaitomas žmonėms, galite naudoti jo ypatybę `__name__`. Pavyzdžiai:

In [22]:

print(type(first_name).__name__)
print(type(is_student).__name__)

str
bool


### Greita Užduotis 5


Išspausdinkite kintamūjų tipus iš praeitos uždoties.

In [26]:
age = 69
print(type(age))
address = 777.7
print(type(address))
first_name = "Bronius"
first_name = True
print(type(first_name))
last_name = "Andriukaitis"
last_name =True
print(type(last_name))
print(type(first_name).__name__)
print(type(last_name).__name__)


<class 'int'>
<class 'float'>
<class 'bool'>
<class 'bool'>
bool
bool


In [29]:
print(type(age))
print(type(address).__name__, type(first_name).__name__, type(is_student).__name__)


<class 'int'>
int str bool


---

## Kintamųjų įvardinimo taisyklės ir iššimtys


Pavadindami kintamąjį "Python", turite laikytis tam tikrų taisyklių ir susitarimų. Čia pateikiamos pagrindinės kintamųjų pavadinimų gairės:

1. Kintamųjų pavadinimai turi prasidėti raide `a-z`, `A-Z` arba pabraukimo ženklu `_`. Jie negali prasidėti skaičiumi ar specialiais simboliais. Pavyzdžiai: `name`, `_count_`, `result`. Netinkami kintamųjų pavadinimai, pvz., `1st_place_winner`.

2. Kintamųjų pavadinimuose gali būti raidžių (a–z, A–Z), skaičių (0–9) ir apatinius brūkšnius (_). Juose skiriamos didžiosios ir mažosios raidės, todėl `age` ir `Age` būtų laikomi skirtingais kintamaisiais. Gerų kintamųjų pavadinimų pavyzdžiai: `age`, `student_number`, `total_sum`, `player`.

Čia taip pat yra pavyzdys, kaip lengva padaryti klaidą nesilaikant griežtos įvardijimo konvencijos

In [30]:
age = 25
Age = 77
print(age, Age)

25 77




3. Rezervuoti žodžiai, taip pat žinomi kaip raktiniai žodžiai, yra žodžiai, kurie yra kalbos sintaksės dalis. Jie turi specifinę reikšmę ir paskirtį, todėl negali būti naudojami kaip kintamieji pavadinimai. Pavyzdžiai: `if`, `for`, `while`, `print` ir kt. Tokiems rezervuotiems žodžiams priskyrus kintamuosius, bus nepaisoma šių raktinių žodžių funkcijų ir sugadintos pagrindinės programavimo kalbos funkcijos.

1. Pasirinkite aprašomuosius ir prasmingus kintamųjų pavadinimus, kad pagerintumėte kodo skaitomumą ir suprantamumą. Pavyzdžiui, loginio tipo kintamajam naudokite `is_student`.

1. Norėdami pagerinti kelių žodžių kintamųjų pavadinimus, naudokite mažąsias raides ir apatinius brūkšnius. Šis susitarimas žinomas kaip gyvatės_atvejis. Pavyzdys: `num_of_students`, `first_name`.

1. Nors nereikalaujama, įprasta naudoti visas didžiąsias raides konstantoms, kintamiesiems, kurių reikšmės nesikeis programos vykdymo metu. Pavyzdys: `PI = 3.14` , `DEFAULT_LANGUAGE = 'en-us'`
``
Svarbu pažymėti, kad „Python“ neleidžia kintamųjų pavadinimuose naudoti specialiųjų simbolių, pvz., „$“ arba „#“, ir paprastai patartina laikytis standartinių pavadinimų suteikimo taisyklių, kad išvengtumėte painiavos ir išlaikytumėte kodo nuoseklumą.

### Greita Užduotis 6 (tiriamoji)

Tikslingai pabandykite naudoti rezervuotus žodžius kaip kintamųjų pavadinimus. Stebėkite gaunamus klaidų pranešimus. Tai padės praktiškai suprasti, kodėl šie žodžiai yra rezervuoti.

### Greita Užduotis 7 (tiriamoji)

Paaiškinkite kodėl kodas žemiau grąžina - False?

In [53]:
str = "Some funny text"
    print(type(str) == str)

IndentationError: unexpected indent (4006299731.py, line 2)

💡Pastaba: po tokių eksperimentų verta iš naujo paleisti Jupyter kernelį, nes Python interpretatoriaus elgesys bus suluošintas priklausomai nuo to, kokius kintamųjų pavadinimus bandėte.


### Papildomi patarimai

- Kintamųjų pavadinimai turėtų būti aiškūs, kad kodas būtų lengviau skaitomas. Pavyzdžiui, vietoj bendrinių pavadinimų, pvz., `x` arba `temp`, pasirinkite prasmingesnius pavadinimus, pvz., `counter` arba `temperature`.

- Gera praktika naudoti daugiskaitos daiktavardžius kintamiesiems, žymintiems rinkinius. Pavyzdžiui, naudokite `fruits`, o ne `fruit`, jei jame yra keli vaisiai.

Vadovaudamiesi šiomis pavadinimo gairėmis, jūsų kodas taps suprantamesnis ir lengviau prižiūrimas tiek jums, tiek kitiems, skaitantiems jūsų kodą.

## Apibendrinimo praktinė užduotis

1. Sukurkite kintamąjį ir priskirkite jam skaičių, tada išspausdinkite kintamojo reikšmę

In [1]:
skaicius = 42
print(skaicius)

42


2. Į aukščiau pateiktą kintamąjį pridėkite neigiamą skaičių ir išspausdinkite gautą reikšmę

In [2]:
skaičius = -777
print(skaičius)

-777


3. Sukurkite slankiojo kablelio (float) kintamąjį, atspausdinkite jo reikšmę, tada pridėkite jį prie kintamųjų iš aukščiau pateiktų užduočių ir išspausdinkite rezultatus

In [4]:
# Sukurkime slankiojo kablelio (float) kintamajį
slankus_kintamasis = 7.16

# Atspausdinkime slankiojo kablelio kintamojo reikšmę
print("Slankiojo kablelio kintamojo reikšmė:", slankus_kintamasis)

# Pridėkime jį prie ankstesnių kintamųjų
pirmas_kintamasis = 42
antras_kintamasis = -777

suma = pirmas_kintamasis + antras_kintamasis + slankus_kintamasis

# Išspausdinkime rezultatus
print("Pirmas kintamasis:", pirmas_kintamasis)
print("Antras kintamasis:", antras_kintamasis)
print("Slankus kintamasis pridėtas prie kitų kintamųjų:", suma)

Slankiojo kablelio kintamojo reikšmė: 7.16
Pirmas kintamasis: 42
Antras kintamasis: -777
Slankus kintamasis pridėtas prie kitų kintamųjų: -727.84


4. Išspausdinkite 1-3 užduočių metu sukurtų kintamųjų tipų pavadinimus.

---

## Papildomi resursai

- Papildomos užduotys [LearnPython.org](https://www.learnpython.org/en/Variables_and_Types)
- Oficialus [Python Reference](https://docs.python.org/3/library/index.html). Net jeigu nelengva suprasti iš karto, tai jum padės ateityje.